src/Export/Adapters/OrderNumberAdapter.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FINDOLOGIC\FinSearch\Export\Adapters;
+
+use FINDOLOGIC\Export\Data\Ordernumber;
+use FINDOLOGIC\FinSearch\Utils\Utils;
+use Shopware\Core\Content\Product\ProductEntity;
+
+class OrderNumberAdapter
+{
+    public function adapt(ProductEntity $product): array
+    {
+        return $this->getParentItemOrderNumber($product);
+    }
+
+    private function getParentItemOrderNumber(ProductEntity $product): array
+    {
+        $orderNumbers = [];
+        $orderNumber = $this->getOrderNumber($product);
+
+        if ($orderNumber === null) {
+            return [];
+        }
+
+        $orderNumbers[] = $orderNumber;
+
+        return $orderNumbers;
+    }
+
+    private function getOrderNumber(ProductEntity $product): ?Ordernumber
+    {
+        if (!Utils::isEmpty($product->getProductNumber())) {
+            return new Ordernumber($product->getProductNumber());
+        }
+
+        if (!Utils::isEmpty($product->getEan())) {
+            return new Ordernumber($product->getEan());
+        }
+
+        if (!Utils::isEmpty($product->getManufacturerNumber())) {
+            return new Ordernumber($product->getManufacturerNumber());
+        }
+
+        return null;
+    }
+}
```

Comment:
We will need to export all 3 types of order numbers. Currently, only the product number would be exported.

```suggestion
    private function getOrderNumber(ProductEntity $product): ?Ordernumber
    {
        $ordernumbers = [];
    
        if (!Utils::isEmpty($product->getProductNumber())) {
            $ordernumbers[] = new Ordernumber($product->getProductNumber());
        }

        if (!Utils::isEmpty($product->getEan())) {
            $ordernumbers[] = new Ordernumber($product->getEan());
        }

        if (!Utils::isEmpty($product->getManufacturerNumber())) {
            $ordernumbers[] = new Ordernumber($product->getManufacturerNumber());
        }

        return $ordernumbers;
    }
}
```